user-fund-storage/src/fund_module.rs

```diff
@@ -421,21 +421,39 @@ pub trait FundModule {
     }
 
     fn destroy_max_for_user(&self,
-        amount: &mut BigUint,
+        mut opt_max_amount: Option<&mut BigUint>,
```

Comment:
i do not think you need this input here, even if it is optional. As the only action which can be done with withdraw only is to claim all. Is destroy_max_for_user called anywhere else ?